### DISCUSSION TITLE
MISP NIDS rules Failing to install

### DISCUSSION THREAD

**Discussion by ow111** [Security-Onion-Solutions/securityonion#2951] [Unsupported Versions]
2021-02-12 15:37:29,394 - <ERROR> - failed to parse rule:
Traceback (most recent call last):
  File "/usr/local/bin/idstools-rulecat", line 12, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/idstools/scripts/rulecat.py", line 858, in main
    rules += idstools.rule.parse_fileobj(
  File "/usr/local/lib/python3.9/site-packages/idstools/rule.py", line 305, in parse_fileobj
    rule = parse(buf + line, group)
  File "/usr/local/lib/python3.9/site-packages/idstools/rule.py", line 245, in parse
    raise Exception("end of option not found")
Exception: end of option not found

Running SO v2.3.21 Distributed deployment.
Getting this issue when running the sudo securityonion-misp/so-misp-setup command. as well as the sudo so-rule-update command. Running them both on the manager node after cloning the git repo, Only thing i can think of is that Suricata is not installed. 
The intel file seems ok on the forward node and the misp data is in there, just the NIDS rules having problems installing.
Confirmed this by doing a grep -i misp /opt/so/rules/nids/all.rules and there were only 3 hits, which don't seem misp related.

**Comment by 0nnyx** [comment]
I had the same problem. It was due to badly written suricata rules in misp events or rules intended for snort. So either you fix these IDS rules or you stick with Zeek intel which works fine. 
idstools is also not really clever in rules processing as it fails and exits at the 1st wrongly formatted rule instead of skipping those rules and processing the good ones.

**Reply by wildzealot** [reply]
You say Zeek Intel works fine? Mine keeps removing the rule file locations that I add to local.zeek.

**Reply by 0nnyx** [reply]
If it keeps removing sthing, it's likely due to changing the zeek running configuration instead of salt one.
My config looks like this : 
` /opt/so/saltstack/local/salt/zeek/policy/intel/__load__.zeek`

```
@load frameworks/intel/seen
@load frameworks/intel/do_notice
@load frameworks/files/hash-all-files
redef Intel::read_files += {
        "/opt/zeek/share/zeek/policy/intel/intel.dat",
        "/opt/zeek/share/zeek/policy/intel/misp-intel.dat"
};
```
The misp-intel.dat is in that same local salt intel directory, updated daily and I get matches without touching anything manually. I also have the otx intel from Wes working properly.

**Reply by TOoSmOotH** [reply]
https://github.com/Security-Onion-Solutions/securityonion/discussions/3178

**Comment by wildzealot** [comment]
Makes sense. Thanks.
